src/vs/editor/common/dnd.ts

```diff
@@ -3,8 +3,18 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { URI } from 'vs/base/common/uri';
+
+export interface IDataTransferFile {
+	readonly name: string;
+	readonly uri?: URI;
+	data(): Promise<Uint8Array>;
+}
+
 export interface IDataTransferItem {
```

Comment:
Maybe this should be a union type now since `asString` returns an empty string on files and `asFile` is undefined on strings

Reply:
Removed `kind` since it was no longer needed